independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/YearParamTest.java

```diff
@@ -0,0 +1,104 @@
+package org.jboss.resteasy.reactive.server.vertx.test.simple;
+
+import java.time.Year;
+import java.time.format.DateTimeFormatter;
+
+import jakarta.ws.rs.CookieParam;
+import jakarta.ws.rs.FormParam;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.QueryParam;
+
+import org.hamcrest.Matchers;
+import org.jboss.resteasy.reactive.DateFormat;
+import org.jboss.resteasy.reactive.server.vertx.test.framework.ResteasyReactiveUnitTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.restassured.RestAssured;
+
+public class YearParamTest {
+
+    @RegisterExtension
+    static ResteasyReactiveUnitTest test = new ResteasyReactiveUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(HelloResource.class, CustomDateTimeFormatterProvider.class));
+
+    @Test
+    public void localDateAsQueryParam() {
+        RestAssured.get("/hello?date=01984")
+                .then().statusCode(200).body(Matchers.equalTo("hello#1984"));
+    }
+
+    @Test
+    public void localDateAsPathParam() {
+        RestAssured.get("/hello/-21")
+                .then().statusCode(200).body(Matchers.equalTo("hello@-21"));
```

Comment:
I also do not see how -21 is a valid year

Reply:
It is

Reply:
ok now I'm suprised by the time api.

Reply:
if I understood this correctly "-21" is passed to `YearParamConverter.convert`, and the parser fails with that input.

Reply:
Try `Year.parse("-21")` in jshell and all your doubts will be dispelled

Reply:
ok I tried it with `PARSER`, my bad